### ISSUE TITLE
pass `mapUrl` function to the defaultRouter

### ISSUE THREAD

**Issue by pSapien** [Bhoos/react-kit#30]
#### DESCRIPTION

- In the present implementation of the `router` (v2.0.11), the `defaultRouter` has no way of knowing the `mapUrl` function unless directly manipulating the object itself. Due to the absence of the `mapUrl` function, the application has no way of redirecting url to its children routers.

**Comment by pSapien** [comment]
```typescript
import { createContext } from 'react';
import { EffectHandler } from '../EffectHandler.js';
import { Route } from './Route.js';

export type ConfirmTransition = (transition: Transition) => void;
type UrlMapper = (url: string, cb: (url: string) => void) => Route | null;

export class Router {
  ....

  constructor(mapUrl?: UrlMapper, stackSize: number = 32) {
    ....
  }
  
  /** something like this maybe? */
  public setMapUrl(mapUrlFn: UrlMapper) {
      this.mapUrl = mapUrlFn;
  }
}
```

**Comment by RazzibShakya** [comment]
Already been resolved in pr #32
